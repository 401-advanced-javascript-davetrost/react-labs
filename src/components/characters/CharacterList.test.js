import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from './CharacterList';


const characters = [
  {
    img: 'https://rickandmortyapi.com/api/character/avatar/327.jpeg',
    name: 'Slow Mobius',
    species: 'Humanoid',
  },
  {
    img: 'https://rickandmortyapi.com/api/character/avatar/370.jpeg',
    name: 'Two Guys with Handlebar Mustaches',
    species: 'Human',
  }
];

describe('Character List component', () => {
  it('renders Character List', () => {
    const wrapper = shallow(<CharacterList characters={characters}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
