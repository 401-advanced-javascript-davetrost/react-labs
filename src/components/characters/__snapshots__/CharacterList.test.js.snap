// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character List component renders Character List 1`] = `
<div
  className="row"
>
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/327.jpeg",
        "name": "Slow Mobius",
        "species": "Humanoid",
      }
    }
    key="0"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/370.jpeg",
        "name": "Two Guys with Handlebar Mustaches",
        "species": "Human",
      }
    }
    key="1"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/155.jpeg",
        "name": "Harold",
        "species": "Cronenberg",
      }
    }
    key="2"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/426.jpeg",
        "name": "Greasy Grandma",
      }
    }
    key="3"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/487.jpeg",
        "name": "Visor Rick",
        "species": "Human",
      }
    }
    key="4"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/265.jpeg",
        "name": "Pickle Rick",
        "species": "unknown, Pickle",
      }
    }
    key="5"
  />
  <Character
    character={
      Object {
        "img": "https://rickandmortyapi.com/api/character/avatar/424.jpeg",
        "name": "Pizza-person",
        "species": "Humanoid, Pizza",
      }
    }
    key="6"
  />
</div>
`;
